uwblueprint/headsupguys
# [Module Builder] Saving and naming a Module

## Notion ticket link

[[Module Builder] Saving and naming a Module](https://www.notion.so/uwblueprintexecs/Module-Builder-Saving-and-naming-a-Module-07351349115b40ebb475a255b6434e05)

## Implementation description
* Navigation from module card to builder page for a module
* Builder page creates new module when name is saved
* Update existing module in builder page with PATCH

## Steps to test
1. Run app locally + connect to database
2. Go to `/admin/dashboard/modules`
3. Click create new module to go to `/admin/dashboard/builder`
4. Enter a name (should give you an error if not provided) and save, the **moduleId should append to the url**
5. Check that the new module shows up in `/admin/dashboard/modules`
6. Click on the card to make sure it redirects to `/admin/dashboard/builder?moduleId=______`

## What should reviewers focus on?
* Modules are correctly saving to database and can be accessed through modules page
* Correct API route is called depending on existence of moduleId in the url (see `handleSaveModule` in `builder.tsx`)

## Checklist
